Microsoft.Toolkit.Uwp/Helpers/SystemInformation.cs

```diff
@@ -76,6 +80,144 @@ public static class SystemInformation
         public static string DeviceManufacturer { get; }
 
         /// <summary>
+        /// Gets a value indicating whether the app is being used for the first time since it was installed.
+        /// Use this to tell if you should do or display something different for the app's first use.
+        /// </summary>
+        public static bool IsFirstRun { get; }
+
+        /// <summary>
+        /// Gets a value indicating whether the app is being used for the first time since being upgraded from an older version.
+        /// Use this to tell if you should display details about what has changed.
+        /// </summary>
+        public static bool IsAppUpdated { get; }
+
+        /// <summary>
+        /// Gets the first version of the app that was installed.
+        /// This will be the current version if a previous verison of the app was installed before accessing this property.
+        /// </summary>
+        public static string FirstVersionInstalled { get; }
+
+        /// <summary>
+        /// Gets the DateTime (in UTC) that the app as first used.
+        /// </summary>
+        public static DateTime FirstUseTime { get; }
+
+        /// <summary>
+        /// Gets the DateTime (in UTC) that this was previously launched.
+        /// Will be DateTime.MinValue if `TrackAppUse` has not been called.
+        /// </summary>
+        public static DateTime LastLaunchTime { get; private set; }
+
+        /// <summary>
+        /// Gets the number of times the app has been launched.
+        /// Will be zero if `TrackAppUse` has not been called.
+        /// </summary>
+        public static long LaunchCount { get; private set; }
+
+        /// <summary>
+        /// Gets the DateTime (in UTC) that this instance of the app was launched.
+        /// Will be DateTime.MinValue if `TrackAppUse` has not been called.
+        /// </summary>
+        public static DateTime LaunchTime { get; private set; }
+
+        /// <summary>
+        /// Gets the length of time this instance of the app has been running.
+        /// Will be TimeSpan.MinValue if `TrackAppUse` has not been called.
+        /// </summary>
+        public static TimeSpan AppUptime
+        {
+            get
+            {
+                if (LaunchCount > 0)
+                {
+                    var subsessionLength = DateTime.UtcNow.Subtract(_sessionStart).Ticks;
+
+                    ApplicationData.Current.LocalSettings.Values.TryGetValue(nameof(AppUptime), out object uptimeSoFar);
+
+                    return new TimeSpan((long)uptimeSoFar + subsessionLength);
+                }
+                else
+                {
+                    return TimeSpan.MinValue;
```

Comment:
Zero has the possibility that it might be valid if checked as soon as the app starts. MinValue will never be valid and so can always be used as a comparison and indicates the value has not been set without the overhead of considering nullability.